drivers/png/image_loader_png.cpp

```diff
@@ -64,7 +65,7 @@ Ref<Image> ImageLoaderPNG::load_mem_png(const uint8_t *p_png, int p_size) {
 	Ref<Image> img;
 	img.instance();
 
-	Error err = PNGDriverCommon::png_to_image(p_png, p_size, img);
+	Error err = PNGDriverCommon::png_to_image(p_png, p_size == -1 ? (sizeof(size_t) == 4 ? INT32_MAX : INT64_MAX) : p_size, img);
```

Comment:
Again, overflowing the integer size is NOT a feature we use (at least not in such obvious case), we shouldn't make code that does it explicitly to silence a warning pointing us to an actual bug.

`p_size` should be unsigned, and `lossless_unpack_png` should be fixed so that it can't pass a negative size (which should already be the case BTW since it errors if `len < 4` and it passes `len - 4`, so I'm not sure what we're fixing here.

Reply:
Just do something like:
```
ERR_FAIL_COND_V_MSG(p_size < 0, Ref<Image>, "Cannot convert PNG to Image resource with a negative size.");
```

Reply:
I can't put that line there, because Godot crash a little after start in OS_X11::set_icon function
![Zrzut ekranu z 2019-11-18 20-15-18](https://user-images.githubusercontent.com/41945903/69082402-4cbf1f80-0a40-11ea-8831-71f7e115b573.png)
Default value of -1 is set to p_len and is used in a lot of functions using Image with constructor
![Zrzut ekranu z 2019-11-18 20-19-48](https://user-images.githubusercontent.com/41945903/69083195-edfaa580-0a41-11ea-8b1c-8ec4c7744a2c.png)
![Zrzut ekranu z 2019-11-18 20-20-04](https://user-images.githubusercontent.com/41945903/69083202-f18e2c80-0a41-11ea-9261-efa66d7d432f.png)
![Zrzut ekranu z 2019-11-18 20-20-14](https://user-images.githubusercontent.com/41945903/69083211-f521b380-0a41-11ea-9be7-bd92815693e8.png)

Reply:
This needs a dedicated bug report, but this manual overflow is definitely not OK.
`Image(const uint8_t *p_mem_png_jpg, int p_len = -1);` is bogus and should be removed, or changed so that a default length of `-1` means that the size is inferred from the byte array. Currently that's not done, so I don't even understand how `make_icon` works at all.